File: src/metrics/custom.ts
import {ExtendedLeafData} from '../types/data';
import {TreeNode} from '../types/node';
import {calculateEuclideanDistance} from './euclidean';
import {calculateTreeDistance} from './tree';


const EW = 0.8;
const TW = 1.2;

export const calculateCustomDistance = (root: TreeNode, x: ExtendedLeafData, y: ExtendedLeafData): number =>
	EW * calculateEuclideanDistance(x, y) + TW * calculateTreeDistance(root, x, y);

File: src/678-dialog/metrics/custom.py
from metrics.euclidean import calculate_euclidean_distance
from metrics.tree import calculate_tree_distance
from ttypes.data import ExtendedLeafData
from ttypes.node import TreeNode


EW = 0.8
TW = 1.2


def calculate_custom_distance(root: TreeNode, x: ExtendedLeafData, y: ExtendedLeafData) -> float:
    return EW * calculate_euclidean_distance(x, y) + TW * calculate_tree_distance(root, x, y)
